spec/services/hmpps_api/assessment_api_spec.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe HmppsApi::AssessmentApi do
+  let(:api_host) { Rails.configuration.assessment_api_host }
+
+  before do
+    stub_auth_token
+  end
+
+  describe '.get_latest_oasys_date' do
+    let(:offender_no) { 'A9346AC' }
+
+    let(:stub_url) { "#{api_host}/offenders/nomisId/#{offender_no}/assessments/summary?assessmentType=LAYER_3&assessmentStatus=COMPLETE" }
+
+    context 'when there is one completed OASys assessment' do
+      # The last assessment date is the 'completed' field.
+      # The test expects it to be a date object for consistency throughout the code base
+      let(:completed_date) { Date.parse('2012-12-10T16:05:55') }
+
+      before do
+        stub_request(:get, stub_url).to_return(status: 200, body: [
+          {
+            "assessmentId": 9165575,
+            "refAssessmentVersionCode": "LAYER3",
+            "refAssessmentVersionNumber": "1",
+            "refAssessmentId": 4,
+            "assessmentType": "LAYER_3",
+            "assessmentStatus": "COMPLETE",
+            "historicStatus": "CURRENT",
+            "refAssessmentOasysScoringAlgorithmVersion": 3,
+            "assessorName": "One Vass1",
+            "created": "2012-12-10T15:48:30",
+            "completed": "2012-12-10T16:05:55"
+          }].to_json)
+      end
+
+      it 'returns the date the assessment was completed' do
+        expect(described_class.get_latest_oasys_date(offender_no)).to eq(completed_date)
+      end
+    end
+
+    context 'when there are multiple completed OASys assessments' do
+      let(:completed_date) { Date.parse('2012-11-18T13:09:06') }
+
+      before do
+        list_of_all_assessments = []
+        # Creates a list of 10 assessments with random dates for the completed field
+        10.times do
+          list_of_all_assessments.push(
+            "assessmentId": 9105544,
+           "refAssessmentVersionCode": "LAYER3",
+           "refAssessmentVersionNumber": "1",
+           "refAssessmentId": 4,
+           "assessmentType": "LAYER_3",
+           "assessmentStatus": "COMPLETE",
+           "historicStatus": "CURRENT",
+           "refAssessmentOasysScoringAlgorithmVersion": 3,
+           "assessorName": "Oliver Orange",
+           "created": "2012-11-18T13:19:59",
+           "completed": Faker::Date.backward
+            )
+        end
+
+        # Assigns one of the completed field to today's date and shuffles them
+        list_of_all_assessments[5]["completed"] = Time.zone.today
```

Comment:
I think this could be a factory – and then it would reduce noise in this file.

`:assessment_api_response`

Thinking out loud:

```ruby
let(:old_assessments) { build_list(:assessment_api_response, 3) }
let(:latest_assessment) { build(:assessment_api_response, completed: Time.zone.now) }
let(:all_assessments) { (old_assessments + [latest_assessment]).shuffle }
# => [ {}, {}, {} ]
```

Reply:
Useful factory which returns a `Hash` object:

`spec/factories/offenders.rb`